### PR TITLE
Workaround: Include web3-provider-engine as a dependency

### PR DESCRIPTION
**Pull request by vishwa-raman**
Issue here for truffle -
https://github.com/oasislabs/truffle/issues/4

There appears to be a change in the way dependencies are packaged in truffle which
causes users to include packages that their sources don't need to get past truffle
compilation issues. For now, we are including web3-provider-engine as a dep here so
that users of c-k don't have to explicitly include this dependency to get past:

Error: Cannot find module 'bindings'

### CODE REVIEW

**Comment by willscott** [comment]
we see this bug down in the web3c dependency -> https://github.com/oasislabs/web3c.js/issues/93 probably want to try to address the bug there if possible.

**Comment by vishwa-raman** [comment]
> we see this bug down in the web3c dependency -> [oasislabs/web3c.js#93](https://github.com/oasislabs/web3c.js/issues/93) probably want to try to address the bug there if possible.

Sure. Will do. Thanks.

**Comment by willscott** [comment]
taking a look to see if we can remove our dependence on truffle-hdwallet-provider entirely, which seems to be the culprit here

**Review by armaniferrante — APPROVED** [review]

**Comment by armaniferrante** [comment]
@vishwa-raman please sign your commit

**Comment by vishwa-raman** [comment]
Will move the dependency per suggestion and then merge. Thanks guys.

Also, will not start a c-k build right now as it runs a confidential contract as part of the tests. Will wait for that to be fixed before re-generating a c-k image.

**Review by willscott — APPROVED** [review]

**Comment by vishwa-raman** [comment]
Can you guys re-approve? I need a re-approval after I moved the dep from dependencies to dev dependencies.

**Review by willscott — APPROVED** [review]

**Comment by armaniferrante** [comment]
Closing and replacing with https://github.com/oasislabs/oasis-box/pull/15.

**Pull request closed without merging**
